packages/vscode-extension/src/commonlib/codeFlowLogin.ts

```diff
@@ -287,6 +294,57 @@ export class CodeFlowLogin {
     }
   }
 
+  async getTokenByScopes(scopes: Array<string>, refresh = true): Promise<Result<string, FxError>> {
+    try {
+      if (!this.account) {
+        const accessToken = await this.login();
+        return ok(accessToken);
+      } else {
+        return this.pca!.acquireTokenSilent({
```

Comment:
Probably not related to this PR (maybe change in a separate PR), but I see many `this.pca!`, `this.scopes!`. Is it better to just declare `pca` as `pca: PublicClientApplication` because `pca` can never be `undefined`. It is initialized in constructor. I think it is better to avoid unsafe operations like this as much as possible.

Reply:
Fixed.